### ISSUE TITLE
Detect electron-prebuilt-compile version through package.json

### ISSUE THREAD

**Issue by jacobq** [electron-userland/electron-packager#909]
Why can't the code in [`infer.js`'s `getVersion`](https://github.com/electron-userland/electron-packager/blob/c0c801465921db516605a1fb5c843f640c54dbb4/infer.js#L41) be replaced with the following? This would make it much easier to use a fork of `electron-prebuilt-compile` (which is frequently several versions behind the latest electron release). 

```js
const electronVersion = require('electron-prebuilt-compile/package.json').version
```

There's a comment saying
> `electron-prebuilt-compile` cannot be resolved because `main` does not point to a valid JS file.

but I'm not sure what that's supposed to mean. Applying the above change in [a fork of this](https://github.com/jacobq/electron-packager/blob/aa55838dc0ebbf3e279103dbc8c4b1f8e242c288/infer.js#L44-L45)  seems to work fine.

**Comment by malept** [comment]
> Why can't the code in [`infer.js`'s `getVersion`](https://github.com/electron-userland/electron-packager/blob/c0c801465921db516605a1fb5c843f640c54dbb4/infer.js#L41) be replaced with the following? This would make it much easier to use a fork of `electron-prebuilt-compile` (which is frequently several versions behind the latest electron release).
> 
> ```js
> const electronVersion = require('electron-prebuilt-compile/package.json').version
> ```

That's because it won't work if you're using the global version of Electron Packager, as opposed to the one installed in a project via `devDependencies`.

> There's a comment saying
> 
> > `electron-prebuilt-compile` cannot be resolved because `main` does not point to a valid JS file.
> 
> but I'm not sure what that's supposed to mean. Applying the above change in [a fork of this](https://github.com/jacobq/electron-packager/blob/aa55838dc0ebbf3e279103dbc8c4b1f8e242c288/infer.js#L44-L45) seems to work fine.

See: https://github.com/electron-userland/electron-packager/pull/608

**Comment by jacobq** [comment]
Would it be acceptable to try to read it from `node_modules` like this first and fallback to the way it currently is if that fails?

**Comment by malept** [comment]
I'd rather not. I have an idea which I may get to this weekend.

**Comment by jacobq** [comment]
OK. The problem I'm actually trying to solve is to be able to specify my own repo for `electron-prebuilt-compile` in an `ember-electron` (`electron-forge` 5.x) project. I've got it hacked together right now with a couple forks & yarn resolutions, and those can go away when `electron-prebuilt-compile` ships 3.0.10, but this isn't the first time I've had to do something like this, and I always wish there was a more orthodox way to do it.
